tools/ci-build.sh

```diff
@@ -63,120 +76,144 @@ P4C_DEPS="bison \
-# These packages are necessary because of a protobuf version mismatch in more recent Ubuntu distributions.
-if [[ "${DISTRIB_RELEASE}" == "22.04" ]] ; then
-  sudo pip3 install --upgrade protobuf==3.20.1
-  sudo pip3 install --upgrade googleapis-common-protos==1.50.0
-  sudo pip3 install --upgrade grpcio==1.51.1
+if [[ "${INSTALL_BMV2}" == "ON" ]] ; then
+  build_bmv2
 fi
+# ! ------  END BMV2 -----------------------------------------------
+
+# ! ------  BEGIN EBPF -----------------------------------------------
+function build_ebpf() {
+  P4C_EBPF_DEPS="libpcap-dev \
+                 libelf-dev \
+                 zlib1g-dev \
+                 llvm \
+                 clang \
+                 iproute2 \
+                 iptables \
+                 net-tools"
+
+  sudo apt-get install -y --no-install-recommends ${P4C_EBPF_DEPS}
+}
 
-# ! ------  BEGIN PTF_EBPF -----------------------------------------------
 function install_ptf_ebpf_test_deps() (
-  P4C_PTF_PACKAGES="gcc-multilib \
-                           python3-six \
-                           libgmp-dev \
-                           libjansson-dev"
-  sudo apt-get install -y --no-install-recommends ${P4C_PTF_PACKAGES}
-
-  git clone --depth 1 --recursive --branch v0.3.1 https://github.com/NIKSS-vSwitch/nikss /tmp/nikss
-  pushd /tmp/nikss
-  ./build_libbpf.sh
-  mkdir build
-  cd build
-  cmake -DCMAKE_BUILD_TYPE=Release ..
-  make "-j$(nproc)"
-  sudo make install
-
-  # install bpftool
-  git clone --recurse-submodules https://github.com/libbpf/bpftool.git /tmp/bpftool
-  cd /tmp/bpftool/src
-  make "-j$(nproc)"
-  sudo make install
-  popd
+    P4C_PTF_PACKAGES="gcc-multilib \
+                             python3-six \
+                             libgmp-dev \
+                             libjansson-dev"
+    sudo apt-get install -y --no-install-recommends ${P4C_PTF_PACKAGES}
+
+    git clone --depth 1 --recursive --branch v0.3.1 https://github.com/NIKSS-vSwitch/nikss /tmp/nikss
+    pushd /tmp/nikss
+    ./build_libbpf.sh
+    mkdir build
+    cd build
+    cmake -DCMAKE_BUILD_TYPE=Release ..
+    make "-j$(nproc)"
+    sudo make install
+
+    # install bpftool
+    git clone --recurse-submodules https://github.com/libbpf/bpftool.git /tmp/bpftool
+    cd /tmp/bpftool/src
+    make "-j$(nproc)"
+    sudo make install
+    popd
 )
-# ! ------  END PTF_EBPF -----------------------------------------------
 
-if [[ "${INSTALL_PTF_EBPF_DEPENDENCIES}" == "ON" ]] ; then
-  install_ptf_ebpf_test_deps
+if [[ "${INSTALL_EBPF}" == "ON" ]] ; then
+  build_ebpf
+  if [[ "${INSTALL_PTF_EBPF_DEPENDENCIES}" == "ON" ]] ; then
+    install_ptf_ebpf_test_deps
+  fi
 fi
+# ! ------  END EBPF -----------------------------------------------
+
+# ! ------  BEGIN DPDK -----------------------------------------------
+function build_dpdk() {
+  # Replace existing Protobuf to match the one specified in the runtime shell.
+  # TODO: Do we really need the shell?
```

Comment:
p4runtime-shell package is needed if you want to run this PTF test without modifying it: https://github.com/p4lang/p4c/blob/main/testdata/p4_16_samples/pna-dpdk-add_on_miss0.py#L27-L29

It could be modified so that it does not use p4runtime-shell, of course, but not clear to me that is worth any effort to make such a change.

Reply:
I believe #4173 will fix this.